client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -344,6 +368,14 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       }
     }
 
+    def getInitialLocs(shuffleId: Int): Array[PartitionLocation] = {
```

Comment:
nit: We return an array which is almost always resized to a much smaller size due to a suffix filter.
Do we want to pass that filter here ? Or return a Seq which is then converted to `toArray` after the filter is applied ?

Reply:
I think it is better to pass a filter.